File: Scala-Project/Task_3_TicketNumber.scala
import scala.io.StdIn._
object Task_3_TicketNumber {

  def main(args: Array[String]): Unit = {
    var numberOfDigit = 0
    var removedLastDigit= 0
    var ticketNumberAfterRemoved = 0
    var remainderTicketNumber=0
    println("Enter the ticket number : Six-Digits only :)")
    var ticketNumber = readInt()
    val temp = ticketNumber
    while(ticketNumber!=0){
      ticketNumber/=10
      numberOfDigit+=1
    }
    ticketNumber = temp
    if(numberOfDigit==6){
      removedLastDigit = ticketNumber%10
      ticketNumberAfterRemoved = ticketNumber/10
      remainderTicketNumber = ticketNumberAfterRemoved%7

      if(remainderTicketNumber == removedLastDigit)
        println("Valid Ticket Number ;)")
      else
        println("Invalid Ticket Number !!!")
    }
    else
      println("Enter six digits please!!!")
  }
}

File: Java-Project/Task_3_TicketNumber.java
import java.util.*;
public class Task_3_TicketNumber {

    public static void main(String[] args) {
        int ticketNumber,numberOfDigits = 0,removedLastDigit,ticketNumberAfterRemoved,remainderTicketNumber,temp;
        Scanner aa = new Scanner(System.in);

        System.out.println("Enter the ticket number : Six-Digits Only :)");
        ticketNumber = aa.nextInt();
        temp = ticketNumber;
        while (ticketNumber!=0){
            ticketNumber/=10;
            ++numberOfDigits;
        }
        ticketNumber = temp;
        if(numberOfDigits == 6){
            removedLastDigit = ticketNumber%10;
            ticketNumberAfterRemoved = ticketNumber/10;
            remainderTicketNumber = ticketNumberAfterRemoved%7;

            if(removedLastDigit == remainderTicketNumber)
                System.out.println("Valid ticket number ;)");
            else
                System.out.println("Invalid ticket number!");
        }
        else
            System.out.println("Enter six digits please!");
    }
}
